### ISSUE TITLE
Support Geopandas

### ISSUE THREAD

**Issue by wsf1990** [modin-project/modin#671]
Hi guys,
    Dose you have the plan to support geopandas?

**Comment by devin-petersohn** [comment]
Hi @wsf1990, thanks for the question!

I would certainly be open to exposing a `modin.geopandas`, if it makes sense for your use case. There was some discussion a few years back on Geopandas about adding Dask: https://github.com/geopandas/geopandas/issues/461, so it might make sense to add a discussion about Modin if you're particularly interested in it. It would be good to gauge interest before a big undertaking.

I will add this topic to the project suggestions meta issue #658. It isn't likely we will personally have time add this in the near future, but contributions and input are always welcome.

**Comment by bubjanes** [comment]
Does that mean it doesn't work with geopandas?

**Comment by janwendt** [comment]
Yes i'm getting Errors when trying to use Geopandas when modin is imported.

**Comment by devin-petersohn** [comment]
@janwendt Geopandas is not yet supported, we are still excited about the potential here!

**Comment by knaaptime** [comment]
hi, just wanted to say thanks for the package and lend one more vote to the request for geopandas support. Fwiw, there has been a pretty large development effort toward [`dask-geopandas`](https://github.com/geopandas/dask-geopandas) lately, and (from what i understand) a lot of the heavy lifting yet to do is around how to intelligently partition spatial data to (a) maximize efficiency since the same number of rows can have very different memory footprints because of very different geometries, and (b) handle partition-spanning operations like computing [spatial weights matrices](https://geographicdata.science/book/notebooks/04_spatial_weights.html) or a [spatial lag](https://github.com/geopandas/dask-geopandas/issues/40). Again, i know there's been lots of effort towards those ends lately, but maybe its also a place where modin's more flexible paritioning ability could come in handy? Anyway, thanks again.

**Comment by mvashishtha** [comment]
Hi all,

I know it's ben a long time since there's been activity on this issue, but I would like to get an understanding of how much interest there is in this feature. Please 👍 this comment or comment if you would like geopandas support.

Alos, does anyone have an idea of how different geopandas is from pandas? How many methods would Modin need to implement?

**Comment by knaaptime** [comment]
hey @mvashishtha thanks for circling back. Personally, I would still be very interested in seeing geopandas support in modin! :)

I dont think it's a trivial lift though. Geopandas is an extension to pandas that wraps spatial operations using [pygeos](https://github.com/pygeos/pygeos) and [shapely](https://github.com/shapely/shapely). I think the critical thing for a project like modin is intelligently handling the spatial index  


cc: @martinfleis @jorisvandenbossche

**Comment by martinfleis** [comment]
I am not familiar with modin but I imagine that to require quite some effort to make it work. One thing are the operations that can be conceptualised as a loop over scalars (though it is vectorized using pygeos/shapely), which can be relatively straightforward. The larger issue is when you need to take into account spatial relationship between individual geometries, where the conceptual overhead of parallel vs single-thread computation is significant. 

But as I said, I have never used modin so I am kind of guessing what you may face.
